### PR TITLE
Enable go test -race

### PR DESCRIPTION
**Pull request by marshall-lee**
This PR enables `-race` option for `go test` and adds some racy tests emulating some load.

### CODE REVIEW

**Comment by benarent** [comment]
@awly Can you review when you get a chance?

File: access/slack/fake_slack_test.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+		values, err := url.ParseQuery(string(data))
+		panicIf(err)
+
+		var msgBlocks slack.Blocks
+		if blocksRaw := values.Get("blocks"); blocksRaw != "" {
+			unescaped, err := url.QueryUnescape(blocksRaw)
+			panicIf(err)
+			err = json.Unmarshal([]byte(unescaped), &msgBlocks)
+			panicIf(err)
+		}
```
Comment (awly) [inline access/slack/fake_slack_test.go:68]: are `blocks` always required?
if yes, make it an error when `blocks` are empty

File: access/slack/fake_slack_test.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+	router.POST("/_response/:ts", func(rw http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+		var payload slack.Msg
+		err := json.NewDecoder(r.Body).Decode(&payload)
+		panicIf(err)
+
+		timestamp := ps.ByName("ts")
+		msg, found := s.GetMessage(timestamp)
+		if !found {
+			_, err = rw.Write([]byte(`{"ok":false}`))
```
Comment (awly) [inline access/slack/fake_slack_test.go:129]: should this also set set StatusNotFound, same as `chat.update` ?

File: access/slack/fake_slack_test.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+func (s *FakeSlack) Close() {
+	s.srv.Close()
+	close(s.newMessages)
+	close(s.messageUpdatesByAPI)
+	close(s.messageUpdatesByResponding)
+}
+
+func (s *FakeSlack) StoreMessage(msg slack.Msg) slack.Msg {
+	if msg.Timestamp == "" {
+		now := s.startTime.Add(time.Now().Sub(s.startTime)) // get monotonic timestamp
```
Comment (awly) [inline access/slack/fake_slack_test.go:185]: how is this different from `time.Now()`?

File: access/slack/slack_test.go [hunk 4]
```diff
@@ -34,18 +38,20 @@ const (
 	HostID      = "00000000-0000-0000-0000-000000000000"
 	Site        = "local-site"
 	SlackSecret = "f9e77a2814566fe23d33dee5b853955b"
+	RaceNumber  = 32
```
Comment (awly) [inline access/slack/slack_test.go:41]: why is this different from other tests?

File: access/slack/slack_test.go [hunk 5]
```diff
@@ -373,33 +380,215 @@ func (s *SlackSuite) TestDenyReadonly(c *C) {
...
+						return setRaceErr(lastErr)
+					}
+					return setRaceErr(trace.Wrap(err))
+				}
+				if err := resp.Body.Close(); err != nil {
+					return setRaceErr(trace.Wrap(err))
+				}
+				if status := resp.StatusCode; status != http.StatusOK {
+					lastErr = trace.Errorf("got %v http code from webhook server", status)
+					runtime.Gosched() // time.Sleep is not good here since we're simulating a tight race.
```
Comment (awly) [inline access/slack/slack_test.go:522]:
```suggestion
```

**Review by awly — APPROVED** [review]

**Comment by benarent** [comment]
@marshall-lee I'm just doing a PR follow up, if this is still relevant can you address Andrews comments. If not, can you close the PR.

**Pull request merged**